src/react/components/common/cloudFilePicker/cloudFilePicker.tsx

```diff
@@ -0,0 +1,172 @@
+import React from "react";
+import { Button, Modal, ModalBody, ModalFooter, ModalHeader } from "reactstrap";
+import { ICloudConnection, IConnection } from "../../../../models/applicationState";
+import { IStorageProvider, StorageProviderFactory } from "../../../../providers/storage/storageProvider";
+import CondensedList, { ListItem } from "../condensedList/condensedList";
+
+export interface ICloudFilePickerProps {
+    isOpen: boolean;
+    connections: IConnection[];
+    onCancel: () => void;
+    onSubmit: (content: string) => void;
+    fileExtension?: string;
+}
+
+export interface ICloudFilePickerState {
+    modalHeader: string;
+    condensedList: any;
+    selectedConnection: ICloudConnection;
+    selectedFile: string;
+    okDisabled: boolean;
+    backDisabled: boolean;
+}
+
+export function isCloudConnection(connection: IConnection): boolean {
+    return "accountName" in connection.providerOptions &&
+            "containerName" in connection.providerOptions;
+}
+
+export class CloudFilePicker extends React.Component<ICloudFilePickerProps, ICloudFilePickerState> {
+
+    constructor(props) {
+        super(props);
+
+        this.getInitialState = this.getInitialState.bind(this);
+        this.handleOk = this.handleOk.bind(this);
+        this.handleBack = this.handleBack.bind(this);
+        this.handleClose = this.handleClose.bind(this);
+        this.connectionList = this.connectionList.bind(this);
+        this.onClickConnection = this.onClickConnection.bind(this);
+        this.fileList = this.fileList.bind(this);
+        this.onClickFile = this.onClickFile.bind(this);
+
+        this.state = this.getInitialState();
+    }
+
+    public render() {
+        const closeBtn = <button className="close" onClick={this.handleClose}>&times;</button>;
+
+        return(
+            <div>
+                <Modal isOpen={this.props.isOpen} centered={true}>
+                    <ModalHeader toggle={this.props.onCancel} close={closeBtn}>
+                        {this.state.modalHeader}
+                    </ModalHeader>
+                    <ModalBody>
+                        {this.state.condensedList}
+                    </ModalBody>
+                    <ModalFooter>
+                        {this.state.selectedFile || ""}
+                        <Button
+                            className="btn btn-success mr-1"
+                            onClick={this.handleOk}
+                            disabled={this.state.okDisabled}>
+                            Ok
```

Comment:
I would rename this to `Open Project`

Reply:
I'll just rename this to `ok`. I want to keep this context neutral just in case there is another use case for browsing cloud files (logs, exports, etc.)